WalletWasabi/Tor/TorProcessManager.cs

```diff
@@ -60,82 +62,102 @@ public TorProcessManager(TorSettings settings, EndPoint torSocks5EndPoint)
+					return false;
+				}
 
-						string torArguments = Settings.GetCmdArguments(TorSocks5EndPoint);
+				string torArguments = Settings.GetCmdArguments(TorSocks5EndPoint) + $" --Log \"notice file {Settings.LogFilePath}\"";
 
-						if (Settings.LogFilePath is { })
-						{
-							IoHelpers.EnsureContainingDirectoryExists(Settings.LogFilePath);
-							var logFileFullPath = Path.GetFullPath(Settings.LogFilePath);
-							torArguments += $" --Log \"notice file {logFileFullPath}\"";
-						}
+				var startInfo = new ProcessStartInfo
+				{
+					FileName = Settings.TorBinaryFilePath,
+					Arguments = torArguments,
+					UseShellExecute = false,
+					CreateNoWindow = true,
+					RedirectStandardOutput = true,
+					WorkingDirectory = Settings.TorDir
+				};
+
+				if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+				{
+					var env = startInfo.EnvironmentVariables;
 
-						var startInfo = new ProcessStartInfo
-						{
-							FileName = Settings.TorBinaryFilePath,
-							Arguments = torArguments,
-							UseShellExecute = false,
-							CreateNoWindow = true,
-							RedirectStandardOutput = true,
-							WorkingDirectory = Settings.TorDir
-						};
-
-						if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-						{
-							var env = startInfo.EnvironmentVariables;
+					env["LD_LIBRARY_PATH"] = !env.ContainsKey("LD_LIBRARY_PATH") || string.IsNullOrEmpty(env["LD_LIBRARY_PATH"])
+						? Settings.TorBinaryDir
+						: Settings.TorBinaryDir + Path.PathSeparator + env["LD_LIBRARY_PATH"];
 
-							env["LD_LIBRARY_PATH"] = !env.ContainsKey("LD_LIBRARY_PATH") || string.IsNullOrEmpty(env["LD_LIBRARY_PATH"])
-								? Settings.TorBinaryDir
-								: Settings.TorBinaryDir + Path.PathSeparator + env["LD_LIBRARY_PATH"];
+					Logger.LogDebug($"Environment variable 'LD_LIBRARY_PATH' set to: '{env["LD_LIBRARY_PATH"]}'.");
+				}
 
-							Logger.LogDebug($"Environment variable 'LD_LIBRARY_PATH' set to: '{env["LD_LIBRARY_PATH"]}'.");
-						}
+				TorProcess = new ProcessAsync(startInfo);
 
-						TorProcess = new ProcessAsync(startInfo);
+				Logger.LogInfo($"Starting Tor process ...");
+				TorProcess.Start();
 
-						Logger.LogInfo($"Starting Tor process ...");
-						TorProcess.Start();
+				if (ensureRunning)
+				{
+					int i = 0;
+					while (true)
+					{
+						i++;
+
+						bool isRunning = await IsTorRunningAsync(TorSocks5EndPoint).ConfigureAwait(false);
 
-						if (ensureRunning)
+						if (isRunning)
 						{
-							Task.Delay(3000).ConfigureAwait(false).GetAwaiter().GetResult(); // dotnet brainfart, ConfigureAwait(false) IS NEEDED HERE otherwise (only on) Manjuro Linux fails, WTF?!!
-							if (!IsTorRunningAsync(TorSocks5EndPoint).GetAwaiter().GetResult())
-							{
-								throw new TorException("Attempted to start Tor, but it is not running.");
-							}
-							Logger.LogInfo("Tor is running.");
+							break;
 						}
+
+						const int MaxAttempts = 25;
+
+						if (i >= MaxAttempts)
```

Comment:
Can `i` be `> MaxAttempts`?

Reply:
I use `>=` in this situations so that a chance of infinite looping is avoided. You are right though.